packages/spor-alert-react-native/src/BaseAlert.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  composeRestyleFunctions,
+  createVariant,
+  spacing,
+  SpacingProps,
+  spacingShorthand,
+  SpacingShorthandProps,
+  useRestyle,
+  VariantProps,
+} from "@shopify/restyle";
+import React, { useEffect, useState } from "react";
+import { Box } from "@vygruppen/spor-layout-react-native";
+import { Theme } from "@vygruppen/spor-theme-react-native";
+import { Text } from "@vygruppen/spor-typography-react-native";
+
+type Variant = VariantProps<Theme, "alertColorScheme", "colorScheme">;
+const colorScheme = createVariant({
+  themeKey: "alertColorScheme",
+  property: "colorScheme",
+});
+
+type RestyleProps = SpacingProps<Theme> &
+  SpacingShorthandProps<Theme> &
+  Variant;
+
+const restyleFunctions = composeRestyleFunctions<Theme, RestyleProps>([
+  spacing,
+  spacingShorthand,
+  colorScheme,
+]);
+
+export type ColorVariants =
+  | "yellow"
+  | "light-yellow"
+  | "orange"
+  | "red"
+  | "green"
+  | "blue";
+
+type AlertProps = Exclude<RestyleProps, "variant"> & {
+  children?: React.ReactNode;
+  colorScheme: ColorVariants;
+  icon: JSX.Element;
+  text: React.ReactNode;
+  expandableItem?: React.ReactNode;
```

Comment:
Jeg ville også endret `icon` til `leftIcon`, og sendt inn evnt lukkeknapp som `rightIcon`. Så kan man sende inn `expandableItem` som `children`:

```tsx
<BaseAlert 
  leftIcon={<InfoIcon />} 
  rightIcon={
    <IconButton 
      icon={<CloseIcon />} 
      onPress={onClose} 
      accessibilityLabel="Lukk" 
    />
  }
>
  {isExpanded ? <Text>Expanded!</Text> : null}
</BaseAlert>
```

Igjen så gir dette litt mer fleksibilitet på "basenivået", mens det gir et litt mindre brukervennlig brukergrensesnitt for direkte konsumenter. Heldigvis er det bare oss selv i `Expandable` og `Closable`-komponentene, så det gjør ikke så mye.